trace_stats/main.cpp

```diff
@@ -152,54 +155,63 @@ void GatherAndPrintStats(const Dive::CaptureMetadata &meta_data, std::ostream &o
+        std::pair(kDirectDraws, "Num Draws (DIRECT)"),
+        std::pair(kTiledDraws, "Num Draws (TILED)"),
+        std::pair(kDispatches, "Num Dispatches"),
+        std::pair(kWaitMemWrites, "Num WaitMemWrites"),
+        std::pair(kWaitForIdle, "Num WaitForIdle"),
+        std::pair(kWaitForMe, "Num WaitForMe"),
+        std::pair(kDepthTestEnabled, "Num Draws with Depth Test Enabled"),
+        std::pair(kDepthWriteEnabled, "Num Draws with Depth Write Enabled"),
+        std::pair(kEarlyZ, "Num Draws with EarlyZ"),
+        std::pair(kLateZ, "Num Draws with LateZ"),
+        std::pair(kEarlyLRZLateZ, "Num Draws with Early LRZ & LateZ"),
+        std::pair(kLrzEnabled, "Num Draws with LRZ Enabled"),
+        std::pair(kLrzWriteEnabled, "Num Draws with LRZ Write Enabled"),
+        std::pair(kCullModeEnabled, "Num Draws with culling enabled"),
+        std::pair(kTotalIndices, "Total indices in all draws (includes non-indexed draws)"),
+        std::pair(kMinIndices, "\tMin indices in a single draw"),
+        std::pair(kMaxIndices, "\tMax indices in a single draw"),
+        std::pair(kMedianIndices, "\tMedian indices in a single draw"),
+        std::pair(kShaders, "Number of unique shaders"),
+        std::pair(kBinningVS, "\tNumber of BINNING VS"),
+        std::pair(kNonBinningVS, "\tNumber of non-BINNING VS"),
+        std::pair(kNonVS, "\tNumber of non-VS Shaders"),
+        std::pair(kTotalInstructions, "Total instructions in all shaders"),
+        std::pair(kMinInstructions, "\tMin instructions in a single shader"),
+        std::pair(kMaxInstructions, "\tMax instructions in a single shader"),
+        std::pair(kMedianInstructions, "\tMedian instructions in a single shader"),
+        std::pair(kTotalGPRs, "Total GPRs in all shaders"),
+        std::pair(kMinGPRs, "\tMin GPRs in a single shader"),
+        std::pair(kMaxGPRs, "\tMax GPRs in a single shader"),
+        std::pair(kMedianGPRs, "\tMedian GPRs in a single shader"),
+        std::pair(kTotalResolves, "Total resolves"),
+        std::pair(kGmemToSysmemResolves, "\tGmem to SysMem Resolves"),
+        std::pair(kGmemToSysmemAndClearGmemResolves, "\tGmem to SysMem Resolves and Clear Gmem"),
+        std::pair(kClearGmemResolves, "\tGmem Clears"),
+        std::pair(kSysmemToGmemResolves, "\tSysMem to Gmem Resolves"),
     };
-#ifndef NDEBUG
-    // Ensure all elements of stats_desc are initialized
-    // Indirect way to ensure the right number of initializers are provided above
-    for (uint32_t i = 0; i < kNumStats; ++i)
-    {
-        assert(stats_desc[i] != nullptr);
-    }
-#endif
+
+    static_assert(kStatMap.size() == kNumStats,
+                  "ERROR: The 'Stat' enum and the 'kStatMap' descriptions are out of sync!");
+
+    constexpr std::array<const char *, kNumStats> kStatDescriptions = [&] {
+        std::array<const char *, kNumStats> arr{};
+        for (const auto &pair : kStatMap)
+        {
+            arr[pair.first] = pair.second;
+        }
```

Comment:
prefer structured bindings here to give more descriptive names (rather than `first` and `second`
```suggestion
        for (const auto &[stat, description] : kStatMap)
        {
            arr[stat] = description;
        }
```